typesense-java/src/test/java/org/typesense/model/CollectionsTest.java

```diff
@@ -0,0 +1,57 @@
+package org.typesense.model;
+
+import junit.framework.TestCase;
+import org.typesense.api.CollectionResponse;
+import org.typesense.api.CollectionSchema;
+import org.typesense.api.Field;
+import org.typesense.resources.Node;
+
+import java.time.Duration;
+import java.util.ArrayList;
+
+public class CollectionsTest extends TestCase {
+
+    public CollectionSchema collectionSchema = new CollectionSchema();
+    public Collections collections;
+    public  Client client;
+
+    public void setUp() throws Exception {
+        super.setUp();
+
+        ArrayList<Node> nodes = new ArrayList<>();
+        nodes.add(new Node("http","localhost","port"));
+
+        Configuration configuration = new Configuration(nodes,Duration.ofSeconds(3),"xyz");
+
+        this.client = new Client(configuration);
+
+    }
+
+    public void testRetrieveAll() {
+        CollectionResponse[] collectionResponses = client.collections().retrieve();
+        for(CollectionResponse c:collectionResponses)
+            System.out.println(c);
+    }
+
+    public void testRetrieveSingleCollection(){
+        System.out.println(client.collections("Countries").retrieve());
```

Comment:
I guess I prefer to use Logger's in the test classes as well, mainly because you can use parameterized strings but also a bit more flexibility around where the logs go if needed. Would prefer to see some Hamcrest/JUnit assertNotNull/assertThat/verify/etc perhaps on the status code if that's available over outputting to stdout, but personal preference.

Reply:
I'm a complete beginner in using these Java libraries. But learning to use them better as we develop this project. Will surely take up your suggestion! : )